mtp_transaction_uploader/upload.py

```diff
@@ -204,6 +206,15 @@ def get_transactions_from_file(data_services_file):
     return transactions
 
 
+def filter_records_from_all_accounts(accounts):
```

Comment:
It took me a second to understand what this did, would a name like `filter_records_from_live_bank_account` self document it better?

Again, not a blocker though!

Reply:
accounts here refers to the file potentially being in multiple-account format as opposed to the single-account format. how about `filter_relevant_records_from_all_accounts` or something? i think _all_ accounts is important, but not sure how to clearly indicate that we're looking for just our transactions

Reply:
Could the code currently handle multiple accounts at the moment? Given it filters out everything but transactions from a single account (or am I reading the code wrong?)

Reply:
single/multiple account is about how the file is structured. the bank can put transactions of multiple accounts into one account in the file (what they currently do) OR they can put transactions of multiple accounts into multiple accounts in the file (a case i want to account for in case they switch ever). it's a weird file format :)

the previous code only handled single account format: `data_services_file.accounts[0].records`, but now it chains all accounts together with `itertools.chain.from_iterable(account.records for account in data_services_file.accounts)`

Reply:
Ah, right I see, so we're getting records from the current file where they have added in transactions from the new account?

Reply:
yeah we are. maybe i can illustrate it…

consider 2 bank accounts A and B and the bank puts transactions involving both into this file. there are 2 formats the file can be in:

**single account format**

```
--account header--
- credit account A
- credit account A
- credit account A
- debit account A
- credit account B
--account trailer--
```

**multiple account format**

```
--account header--
- credit account A
- credit account A
- credit account A
- debit account A
--account trailer--
--account header--
- credit account B
--account trailer--
```

we currently get the file in the first format, but i want to cater for the second format as well. the conversations with the bank have been long chains of emails and i worry they might switch us to the second format because it was mentioned at some point

Reply:
Cool, makes sense now.